modules/admin_manual/pages/configuration/user/oidc/ms-azure-setup.adoc

```diff
@@ -0,0 +1,127 @@
++
+image:configuration/user/oidc/ms-azure-ad/setup-claims-ms-azure-ad.png[,width=90%]
+
+. Go to menu:API permissions[Configured permissions > Add a permission] and add delegated permissions.
++
+image:configuration/user/oidc/ms-azure-ad/graph-permission-ms-azure-ad.png[,width=90%]
+
+. On the same page, do not forget to set menu:Grant admin consent[] to all permissions.
++
+image:configuration/user/oidc/ms-azure-ad/admin-consent-ms-azure-ad.png[,width=90%]
+
+. Go to menu:Expose an API[Set the App ID URI] and set the *Application ID URI*.
++
+image:configuration/user/oidc/ms-azure-ad/app-id-uri-ms-azure-ad.png[,width=90%]
+
+. Then on the same page menu:Add a scope[Scope name] give the scope a meaningful name like *owncloud* and menu:Add a scope[Who can consent] allow *Admins and users* to consent. You will see the full api  [#api-scope]#scope name# below your entered scope name.
++
+image:configuration/user/oidc/ms-azure-ad/scope-consent-ms-azure-ad.png[,width=90%]
+
+. You get the [#client-id]#CLIENT-ID# after you've completed the MS Azure Setup process.
+
+=== Configure ownCloud Server
+
+To set up ownCloud Server to work with OpenID Connect, you have to:
+
+1. Install the {oc-marketplace-url}/apps/openidconnect[OpenID Connect App]
+2. Configure `config.php`
+3. Set up service discovery
+
+TIP: It is recommended to first figure out all configurations on a test system and to bring it to the production system once it's proven to work. Enabling the OpenID Connect App on the production system should be the last step in this process as it will then advertise OpenID Connect to all clients.
+
+==== List of OpenID Connect config.php Parameters
+
+Follow this link to read more about the
+xref:configuration/server/config_apps_sample_php_parameters.adoc#app-openid-connect-oidc[OIDC config.php parameters] available to configure OpenID Connect on ownCloud Server.
+
+==== Example config.php setup
+
+An example snippet that can be added to `config.php` is shown below. You need to add both config values as listed below.
+
+Use these links to see the corresponding configuration section for: 
+
+* xref:tenant-id[YOUR-DIRECTORY-TENANT-ID]
+* xref:client-secret[YOUR-CLIENT-SECRET]
+* xref:api-scope[YOUR-APPLICATION-ID-SCOPE-URI/owncloud]
+* xref:client-id[YOUR-CLIENT-ID]
+
+[source,php]
+----
+'http.cookie.samesite' => 'None',
+
+'openid-connect' => [
+    'provider-url' => 'https://login.microsoftonline.com/YOUR-DIRECTORY-TENANT-ID/v2.0/',
+    'client-id' => 'YOUR-CLIENT-ID',
+    'client-secret' => 'YOUR-CLIENT-SECRET',
+    'loginButtonName' => 'Azure AD',
+    'autoRedirectOnLoginPage' => false,
+    'scopes' => [
+        'openid',
+        'api://YOUR-APPLICATION-ID-SCOPE-URI/owncloud',
+        'profile', 'email', 'offline_access',
+    ],
+    'mode' => 'email',
+    'search-attribute' => 'unique_name',
+    'use-access-token-payload-for-user-info' => true,
+    'auto-provision' => [
```

Comment:
not necessary for azure - actually auto-provision is independent of the idp
should go into the general documentation

explanation:
if auto-provision is setup an owncloud user will be created after successful login using openid connect.
if auto-provision is not setup - it is expected that the user exists - this is where usualy an LDAP setup is required additionally

Reply:
Thanks for the explanation !
Fixed here and PR on the way in core for the config.apps.sample part

Reply:
The core PR is here: https://github.com/owncloud/core/pull/38638